core/shared/src/main/scala/bobcats/Hotp.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bobcats
+
+import cats.effect.kernel.Sync
+import cats.syntax.functor._
+import scodec.bits.ByteVector
+
+sealed trait Hotp[F[_]] extends HotpPlatform[F] {
+  def generate(
+      key: SecretKey[HmacAlgorithm.SHA1.type],
+      movingFactor: Long,
+      digits: Int = 6
```

Comment:
I think that `digits` must really be between 6 and 8, and nothing larger than 10^9 digits first in an `Int` anyway. How would you usually deal with a precondition like that in this project?`require`?

Reply:
Good question. This project is too experimental to have any good standards yet 😂 

Cats Effect uses `require` for these sort of things. In fact since we have `Sync` we could just `raiseError`.

Btw, instead of a default arg an overload would be better here, bincompat and stuff.

Thanks for the PRs btw! :)